lib/model/model_test.go

```diff
@@ -947,11 +947,13 @@ func TestIgnores(t *testing.T) {
 
 	db := db.OpenMemory()
 	m := NewModel(defaultConfig, protocol.LocalDeviceID, "device", "syncthing", "dev", db, nil)
-	m.AddFolder(defaultFolderConfig)
-	m.StartFolder("default")
 	m.ServeBackground()
+	m.cfg.SetFolder(defaultFolderConfig)
 	defer m.Stop()
 
+	// wait for goroutine wrapper.notifyListener to finish
+	time.Sleep(time.Duration(5) * time.Millisecond)
```

Comment:
@calmh
Well that actually isn't reliable as the `configWaiter` is triggered shortly after `model.CommitConfiguration` starts, not when it finished -> racy again. So I tried something else: I added a `ConfigChanged` event emitted at the end of `model.CommitConfiguration` and wait for that event. Well it is not enough, the good old weirdo error turns up again https://build.syncthing.net/job/syncthing-pr/3544/console So back to square one: No clue why it happens. Now we are even sure that the folder was added and started. Is there any way to find out what the program does in the 5ms sleep that are apparently enough to fix it?